### PR TITLE
add Clojure

### PR DESCRIPTION
**Pull request by bhougland18**
This adds clojure via https://github.com/sogaiu/tree-sitter-clojure

However, I am not able to get this to compile and I assume it is because of my unfamiliarity with rust, as it is telling me there is an 'eof' error in the parser.c file.  I hate to ask, but does this work for you?

Also, this is my first time committing to a project so please let me know everything else I did incorrectly and I will do my best to correct it.

### CODE REVIEW

File: .gitmodules [hunk 1]
```diff
@@ -81,3 +81,6 @@
 [submodule "vendor/tree-sitter-elm"]
 	path = vendor/tree-sitter-elm
 	url = https://github.com/Razzeee/tree-sitter-elm.git
+[submodule "vendor/tree-sitter-clojure"]
+	path = vendor/tree-sitter-clojure
+	url = https://github.com/sogaiu/tree-sitter-clojure
```
Comment (Delapouite) [inline .gitmodules]: Looks like `.git` is missing at the end in comparison with lines above.

Comment (bhougland18) [reply .gitmodules]: oh no.  I will correct this now

Comment (bhougland18) [reply .gitmodules]: wait, I thought these lines were added directly from using the git submodule command.  I didn't hand-enter these line.

Comment (bhougland18) [reply .gitmodules]: looks like all the other entries contain the .git type, so I updated the .gitmodules file.  I probably did something wrong.

**Pull request closed without merging**